lib/events.js

```diff
@@ -394,7 +396,28 @@ EventEmitter.prototype.listeners = function listeners(type) {
   return ret;
 };
 
-EventEmitter.listenerCount = function(emitter, type) {
+EventEmitter.prototype.listenerCount = function listenerCount(type) {
+
+  if (typeof type !== 'string') {
+    throw new TypeError('type must be a valid string');
+  }
+
+  const events = this._events;
+
+  if (events && events.hasOwnProperty(type)) {
```

Comment:
Wouldn't it be better to use something like `Object.prototype.hasOwnProperty.call(events, type)` instead?

Also FWIW `hasOwnProperty()` still causes a significant performance hit and would technically not be backwards compatible behavior.